use-cases/Customizing a package definition.md

```diff
@@ -0,0 +1,40 @@
+# Customizing a package definition
+
+## General information
+
+A package definition is code that defines:
+
+- how a package is built from source
+- metadata for the package such as its name, license, or homepage
+- dependencies of the package
+
+Different users use packages differently. They might want to:
+
+- turn optimizations on or off
+- turn features on or of
+- change versions of dependencies
+- include patches
+- add build steps
+- add abstractions to the customization interface
+- update metadata
+
+## Concrete examples
+
+Scenario 1: Adding a patch to a package
+Alice wants to apply a patch to the examplePackage from Nixpkgs to fix a bug that affects her system. She needs a simple way to add the patch to the package definition without having to modify the entire package.
+
+Scenario 2: Changing a build flag
+Bob wants to build examplePackage with a specific feature flag enabled. He should be able to modify the package definition to enable the feature without affecting other users or requiring an entirely new package definition.
```

Comment:
```suggestion
Bob wants to build examplePackage with a specific feature flag enabled. He should be able to modify the package definition to enable the feature without affecting other packages or requiring an entirely new package definition.
```

I think this might describe it better